ton/wallet/wallet.go

```diff
@@ -465,6 +465,32 @@ func (w *Wallet) SendWaitTransaction(ctx context.Context, message *Message) (*tl
 	return w.SendManyWaitTransaction(ctx, []*Message{message})
 }
 
+// TransferWaitTransaction always waits for tx block confirmation and returns found tx.
+func (w *Wallet) TransferWaitTransaction(ctx context.Context, addr *address.Address, amount tlb.Coins, comment string) (*tlb.Transaction, *ton.BlockIDExt, error) {
+	var body *cell.Cell
+	var err error
+
+	if comment != "" {
```

Comment:
Take a look at `w.BuildTransfer` method, we can replace this part with it's call to simplify code, same logic there with comment and message creation